pkg/virt-controller/watch/patchreactor_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 The KubeVirt Authors
+ *
+ */
+
+package watch
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	jsonpatch "github.com/evanphx/json-patch"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/strategicpatch"
+	"k8s.io/client-go/testing"
+)
+
+func PatchReactor(handle func(string) bool, tracker testing.ObjectTracker,
+	modify func(new, old runtime.Object) runtime.Object,
```

Comment:
Would it make sense to create a type for this function signature?

Reply:
I'm wondering for example, also if it would make sense to add an error as return value

Reply:
I will improve the documentation. I don't think error has value as this is only for testing and failing is as good.

Reply:
OK, I think the error could have happen in the modify function if one of the object were nil.

Reply:
It is better to fail in that function than propagate